phac-nml/irida-next
# Delete metadata

## What does this PR do and why?
This PR adds the ability to delete metadata through the website UI. Users with access can delete metadata either by the action dropdown, or by using the checkboxes and clicking the `Delete Metadata` button.

## Screenshots or screen recordings
Delete by action dropdown:
[Screencast from 2024-02-14 08:09:23 AM.webm](https://github.com/phac-nml/irida-next/assets/82407232/8246cb8a-9e1f-45cd-b674-bf318c00bbeb)

Delete multiple fields with checkboxes and modal:
[Screencast from 2024-02-14 08:09:51 AM.webm](https://github.com/phac-nml/irida-next/assets/82407232/0904e84d-b207-44fd-b116-522fe88d6882)

Metadata page for user < Maintainer access:
![image](https://github.com/phac-nml/irida-next/assets/82407232/b6a2dea6-9081-4433-a902-e0d297971a54)


## How to set up and validate locally
1. Navigate to a sample's metadata page
2. Ensure you are able to delete metadata by both the action dropdown and with the checkboxes and `Delete Metadata` button
3. Logic for attachment checkboxes was slightly altered due to the implementation of the metadata checkboxes, so ensure the following:
a) Ensure checking off files and metadata does not enable the other's action buttons (ie: checking off metadata does not enable concatenate/delete files buttons, checking off files does not enable the delete metadata button)
b) Ensure file concatenation behaves as expected
c) Check both files and metadata for deletion. Ensure deleting one set does not affect the other set (ie: deleting files does not affect any metadata)
4. Ensure a user with access < Maintainer does not see metadata checkboxes, action dropdown, or the `Delete Metadata` button.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
